addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/internal/config/NestStructureConfiguration.java

```diff
@@ -8,14 +8,17 @@
  */
 package org.openhab.binding.nest.internal.config;
 
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
 /**
  * The configuration for structures.
  *
  * @author Wouter Born - Initial contribution
  * @author Wouter Born - Add device configuration to allow file based configuration
  */
+@NonNullByDefault
 public class NestStructureConfiguration {
     public static final String STRUCTURE_ID = "structureId";
     /** Structure ID which can be retrieved with the Nest API. */
-    public String structureId;
+    public String structureId = "";
```

Comment:
Will / can it never be set to `null`?

Reply:
It's a required configuration parameter see the Nest [config.xml](https://github.com/wborn/openhab2-addons/blob/d3e2a278b8b101e4b98133b20ed088dac840321a/addons/binding/org.openhab.binding.nest/ESH-INF/config/config.xml#L42).

Without the `structureId` the handler would not know what data from updates should be used for state updates. For the same reason the Nest API needs this ID for handling commands.

I was doubting between annotating it with `@NonNullByDefault({})` and initializing it with an empty string. The empty string option looks cleaner and is guaranteed to prevent NPEs.

Reply:
Okay, great